### DISCUSSION TITLE
Allow documentation of autoloads with GDScript documentation comments

### DISCUSSION THREAD

**Discussion by Pennycook** [godotengine/godot-proposals#7722] [Editor]
I use GDScript documentation comments extensively for my editor plugins.  Unless I'm mistaken, classes only show up in the "Search Help" list if they've been defined with `class_name`, which makes it difficult to document the behavior of autoload singletons.  Attempting to define an autoload with the same name as a class results in an error.

The way I'm currently working around this is to define separate (but related) names: I have a class called `InputPromptManager` which documents that it is autoloaded under the name `PromptManager`.  Although this works, it's not very satisfying, as the behavior is different from built-in singletons (like `InputMap`).

I'm not sure what it would take to make this work, but ideally I'd like an entry for each autoloaded class to show up in the "Search Help" list.

**Comment by dalexeev** [comment]
* See also #8441.
